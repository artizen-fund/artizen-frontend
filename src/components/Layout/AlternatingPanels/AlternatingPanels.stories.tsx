import AlternatingPanels from './'

export default {
  title: 'components/AlternatingPanels',
  component: AlternatingPanels,
  argTypes: {},
}

export const AlternatingPanelsComponent = (props: any) => <AlternatingPanels {...props} />
