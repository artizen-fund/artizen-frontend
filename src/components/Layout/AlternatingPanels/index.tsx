import { cloneElement, Children } from 'react'
import styled from 'styled-components'
import { IAlternatingPanel } from '../AlternatingPanel'
import { PagePadding } from '@components'
import { breakpoint } from '@theme'

interface IAlternatingPanels {
  children: Array<React.ReactElement<IAlternatingPanel>>
}

const AlternatingPanels = ({ children }: IAlternatingPanels) => (
  <PagePadding>
    <Wrapper>{Children.map(children, (child, i) => cloneElement(child, { imageOnRight: !!(i % 2) }))}</Wrapper>
  </PagePadding>
)

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  @media only screen and (min-width: ${breakpoint.tablet}px) {
    gap: 50px;
  }
  @media only screen and (min-width: ${breakpoint.laptop}px) {
    gap: 65px;
  }
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    gap: 85px;
  }
`

export default AlternatingPanels
