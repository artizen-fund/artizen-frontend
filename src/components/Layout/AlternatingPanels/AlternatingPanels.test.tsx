import renderer from 'react-test-renderer'
import AlternatingPanels from './'

describe('AlternatingPanels', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AlternatingPanels />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
