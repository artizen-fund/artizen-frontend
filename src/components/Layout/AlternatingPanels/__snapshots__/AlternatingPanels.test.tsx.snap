// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlternatingPanels renders correctly 1`] = `
<div
  className="sc-9199b4ac-0 brREMd"
>
  <div
    className="sc-d74fb460-0 kscYoG"
  />
</div>
`;
